tests/functional/test_logger_powertools_formatter.py

```diff
@@ -288,3 +290,22 @@ def test_log_formatting(stdout, service_name):
 
     # THEN the formatting should be applied (NB. this is valid json, but hasn't be parsed)
     assert log_dict["message"] == '["foo bar 123 [1, None]", null]'
+
+
+def test_log_json_indent_compact_indent(stdout, service_name, monkeypatch):
+    # GIVEN a logger with default settings and WHEN POWERTOOLS_DEV is not set
+    if "POWERTOOLS_DEV" in os.environ:
+        monkeypatch.delenv(name="POWERTOOLS_DEV")
+    logger = Logger(service=service_name, stream=stdout)
+    logger.info("Test message")
+    # THEN the json should not be indented using constant.PRETTY_INDENT blank spaces
+    assert " " * constants.PRETTY_INDENT not in stdout.getvalue()
+
+
+def test_log_json_pretty_indent(stdout, service_name, monkeypatch):
+    # GIVEN a logger with default settings and WHEN POWERTOOLS_DEV=="true"
+    monkeypatch.setenv(name="POWERTOOLS_DEV", value="true")
+    logger = Logger(service=service_name, stream=stdout)
+    logger.info("Test message")
+    # THEN the json should contain indentation (of constant.PRETTY_INDENT blank spaces)
+    assert " " * constants.PRETTY_INDENT in stdout.getvalue()
```

Comment:
```suggestion
    new_lines = stdout.getvalue().count("\n")
    assert new_lines > 1
```

See my earlier comment. You could also do `assert new_lines == 7` if you want to be 100% precise, but it's brittle if we add additional keys.

Reply:
I've commited your suggestion, but using os.linesep